### PR TITLE
Revendor cluster api

### PR DESCRIPTION
**Pull request by enxebre**
Revendor cluster api to accomodate https://github.com/openshift/cluster-api/pull/34

### CODE REVIEW

**Comment by ingvagabund** [comment]
Let's wait until https://github.com/openshift/cluster-api-actuator-pkg/pull/73 gets merged

/hold

File: pkg/actuators/machine/utils.go [hunk 1]
```diff
@@ -149,34 +147,13 @@ func terminateInstances(client awsclient.Client, instances []*ec2.Instance) erro
 
 // providerConfigFromMachine gets the machine provider config MachineSetSpec from the
 // specified cluster-api MachineSpec.
-func providerConfigFromMachine(client client.Client, machine *machinev1.Machine, codec *providerconfigv1.AWSProviderConfigCodec) (*providerconfigv1.AWSMachineProviderConfig, error) {
-	var providerSpecRawExtention runtime.RawExtension
-
-	providerSpec := machine.Spec.ProviderSpec
-	if providerSpec.Value == nil && providerSpec.ValueFrom == nil {
+func providerConfigFromMachine(machine *machinev1.Machine, codec *providerconfigv1.AWSProviderConfigCodec) (*providerconfigv1.AWSMachineProviderConfig, error) {
+	if machine.Spec.ProviderSpec.Value == nil {
```
Comment (frobware) [inline pkg/actuators/machine/utils.go]: We need to update the error message (next line) - still mentions `ValueFrom`.

**Comment by enxebre** [comment]
```level=error msg="1 error occurred:"
level=error msg="\t* module.vpc.aws_route.igw_route: 1 error occurred:"
level=error msg="\t* aws_route.igw_route: Error creating route: timeout while waiting for state to become 'success' (timeout: 20m0s)"
level=error
level=error
level=error
level=error
level=error
level=error msg="Terraform does not automatically rollback in the face of errors."
level=error msg="Instead, your Terraform state file has been partially updated with"
level=error msg="any resources that successfully completed. Please address the error"
level=error msg="above and apply again to incrementally change your infrastructure."
```
/retest

**Pull request merged**